moto/sagemaker/models.py

```diff
@@ -677,6 +718,58 @@ def get_notebook_instance_tags(self, arn):
         except RESTError:
             return []
 
+    def create_notebook_instance_lifecycle_config(
+        self, notebook_instance_lifecycle_config_name, on_create, on_start
+    ):
+        lifecycle_config = FakeSageMakerNotebookInstanceLifecycleConfig(
+            region_name=self.region_name,
+            notebook_instance_lifecycle_config_name=notebook_instance_lifecycle_config_name,
+            on_create=on_create,
+            on_start=on_start,
+        )
+        self.notebook_instance_lifecycle_configurations[
+            notebook_instance_lifecycle_config_name
+        ] = lifecycle_config
+        return lifecycle_config
+
+    def describe_notebook_instance_lifecycle_config(
+        self, notebook_instance_lifecycle_config_name
+    ):
+        try:
+            return self.notebook_instance_lifecycle_configurations[
+                notebook_instance_lifecycle_config_name
+            ].response_object
+        except KeyError:
+            message = "Unable to describe Notebook Instance Lifecycle Config '{}'. (Details: Notebook Instance Lifecycle Config does not exist.)".format(
+                FakeSageMakerNotebookInstanceLifecycleConfig.arn_formatter(
+                    notebook_instance_lifecycle_config_name, self.region_name
+                )
+            )
+            raise RESTError(
+                error_type="ValidationException",
+                message=message,
+                template="error_json",
+            )
+
+    def delete_notebook_instance_lifecycle_config(
+        self, notebook_instance_lifecycle_config_name
+    ):
+        try:
+            del self.notebook_instance_lifecycle_configurations[
+                notebook_instance_lifecycle_config_name
+            ]
+        except KeyError:
+            message = "Unable to delete Notebook Instance Lifecycle Config '{}'. (Details: Notebook Instance Lifecycle Config does not exist.".format(
+                FakeSageMakerNotebookInstanceLifecycleConfig.arn_formatter(
+                    notebook_instance_lifecycle_config_name, self.region_name
+                )
+            )
+            raise RESTError(
```

Comment:
Let's extract both RESTerror into a ValidationError in exceptions.py, and

Reply:
Done, and changed all existing raises of RESTError in this module to use it.